src/pydata_sphinx_theme/pygments.py

```diff
@@ -99,5 +99,5 @@ def overwrite_pygments_css(app: Sphinx, exception=None):
 
     # re-write pygments.css
     pygments_css = Path(app.builder.outdir) / "_static" / "pygments.css"
-    with pygments_css.open("w") as f:
-        f.write(get_pygments_stylesheet(light_theme, dark_theme))
+    pygments_css.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
I think you shouldn't need `parents=True` here, right?  Isn't `app.builder.outdir` guaranteed to exist?

Reply:
(also please add a comment saying something like "_static folder not guaranteed to exist during linkcheck" or similar)

Reply:
Sorry for the late response.
Yeah, you are right the existence of `app.builder.outdir` by a [check in the init of app](https://github.com/sphinx-doc/sphinx/blob/e7ca2d8eaf047ddb7b860cc16e67c7faf75317ee/sphinx/application.py#L215), which in turn is [used when initializing the builder base class](https://github.com/sphinx-doc/sphinx/blob/e7ca2d8eaf047ddb7b860cc16e67c7faf75317ee/sphinx/builders/__init__.py#L113).
So, as long as no one overwrites the inits without calling the `super` methods in a custom subclass of those, it is guaranteed to exist. 👍